### PR TITLE
Scaling to zero takes too much time in case of non-scaling related runspec changes

### PR DESCRIPTION
**Pull request by meln1k**
Summary: Deployment plan now treats scale to 0 instances change as scale-only change regardless other changes in the runspec

JIRA issues: MARATHON-8011

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala [hunk 1]
```diff
@@ -422,6 +422,36 @@ class DeploymentPlanTest extends UnitTest with GroupCreation {
       Then("The deployment is not valid")
       validate(plan2)(f.validator).isSuccess should be(false)
     }
+
+    "Deployment plan treats change with 0 instances as scale-only change" in {
```
Comment (zen-dog) [inline src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala]: nit: `s/change with 0 instances/change *and* scale down to 0 instances` to make the case clear

File: src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala [hunk 2]
```diff
@@ -422,6 +422,36 @@ class DeploymentPlanTest extends UnitTest with GroupCreation {
...
+      Given("An application update with command and scale changes")
+      val mongoId = "/test/database/mongo".toPath
```
Comment (zen-dog) [inline src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala]: `mongo`? rly? :D

**Comment by zen-dog** [comment]
Don't forget to backport

File: src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala (hunk 2 above)
Comment (meln1k) [reply src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala]: It's a kind of database you'd like to scale to 0 instances as soon as possible :)

Comment (zen-dog) [reply src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala]: oh snap 😂

**Pull request merged**